proposed/2022/PrivateAssetIndepdentFromExcludeAsset.md

```diff
@@ -0,0 +1,231 @@
+# Opt-in experience for PrivateAssets and ExcludeAssets options are independent
+
+- Author Name <https://github.com/erdembayar>
+- Start Date 2022-12-07
+- GitHub Issue <https://github.com/NuGet/Home/issues/6938>
+- GitHub PR <https://github.com/NuGet/NuGet.Client/pull/4976>
+
+## Summary
+
+<!-- One-paragraph description of the proposal. -->
+Currently asset consumption via `PrivateAssets` option experience for assets from transitive package in a project/package is not deterministic for parent consuming project, here `PrivateAssets` option calculation is not independent from `IncludeAssets/ExcludeAssets` option.
+This proposal introduces new a `PrivateAssetIndependent` opt-in property to make `PrivateAssets` option independent from `IncludeAssets/ExcludeAssets` option.
```

Comment:
from https://github.com/NuGet/Home/pull/12313#discussion_r1043946068:

> The name PackPrivateAssetsFlow would need some workshopping given the feedback.

(the name was then changed to `PrivateAssetIndependent`)

I think it's worthwhile thinking of more ideas, because I think `PrivateAssetIndependent` only makes sense to people who have read https://github.com/NuGet/Home/issues/6938. Trying to put myself in the mindset of a customer who has not read that github issue, I don't think most people would guess it's related to which assets get excluded in transitive packages.

I think all my suggests names are too long, but something like `PrivateAssetsDefineTransitiveFlow`? `PrivateAssetsExcludeTransitiveFlow`? `UsePrivateAssetsForDependencies`? `DependenciesFlowViaPrivateAssets`? `TransitiveFlowViaPrivateAssets`? `DependenciesFlowWithPrivateAssetsOnly`? `TransitiveAssetFlowWithPrivateAssets`? If we don't want to limit ourselves to true/false values, maybe `TransitiveAssetFlow` with default value `Calculated`, and opt-in `PrivateAssets`?

I kinda hate all my own suggestions, except the last one, but I'm guessing most people would prefer true/false values, not a custom enum. But I think that without context about what "independent" means, I think they're clearer to understand, more likely that someone who knows nothing about the feture might have a chance of guessing what it means. Although, maybe the name doesn't matter. None of them are super obvious, so customers will probably need to look up docs regardless of what the name is.

Reply:
Thank you for your suggestion. I'll wait if other folks have any preference between `PrivateAssetIndependent` vs `Suggestions` or one of the above suggestions. I welcome anyone can come up with better naming suggestion.

Reply:
How about `ExcludedAssetsMayFlowTransitively`?

Reply:
@NuGet/nuget-client @JonDouglas
Any other suggestions for better naming for new propery?